storagemanager/storagemanager.go

```diff
@@ -22,7 +22,15 @@ var (
 )
 
 type Config struct {
-	MaxStagingDealsBytes uint64
+	MaxStagingDealsBytes          uint64
+	MaxStagingDealsPercentPerHost uint64
+}
+
+func (c Config) check() error {
+	if c.MaxStagingDealsPercentPerHost > 100 {
+		return fmt.Errorf("MaxStagingDealsPercentPerHost is %d but it must be a percentage between 0 - 100", c.MaxStagingDealsPercentPerHost)
+	}
```

Comment:
Nit: I suggest embedding inline this, and getting rid of `check()`. I don't see much value in having it extracted in a separate function.